pull.go

```diff
@@ -130,10 +133,17 @@ func (c *Client) fetch(ctx context.Context, rCtx *RemoteContext, ref string, lim
 			},
 		)
 
+		// add distribution source label to blob data
+		distributionSourceLabelHandler, err := docker.SetDistributionLabels(store, ref)
+		if err != nil {
+			return images.Image{}, err
+		}
+
 		handler = images.Handlers(append(rCtx.BaseHandlers,
 			remotes.FetchHandler(store, fetcher),
 			convertibleHandler,
 			childrenHandler,
+			distributionSourceLabelHandler,
```

Comment:
We might want an option around this handler as well, I don't think the CRI plugin will care much about setting this.

Reply:
sure. will do.